src/niscope/system_tests/test_system_niscope.py

```diff
@@ -195,6 +210,130 @@ def test_fetch_double_into(session):
+    assert uncleared_stats[0].max_val == uncleared_stats_2[0].max_val
+    assert uncleared_stats[0].num_in_stats == uncleared_stats_2[0].num_in_stats
+    assert uncleared_stats[0].num_in_stats != cleared_stats[0].num_in_stats
+
+
+def test_waveform_processing(session):
+    test_voltage = 1.0
+    test_record_length = 1000
+    test_channels = range(2)
+    test_num_channels = 2
+    test_num_records = 3
+    session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
+    session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
+    with session.initiate():
+        session.add_waveform_processing(niscope.enums.ArrayMeasurement.DERIVATIVE)
+        processed_waveforms = session.channels[test_channels].fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
+        session.clear_waveform_processing()
+        unprocessed_waveforms = session.channels[test_channels].fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
+
+    assert len(processed_waveforms) == test_num_channels * test_num_records
+    assert len(unprocessed_waveforms) == test_num_channels * test_num_records
+    # Here the idea is to leave a large margin to not test too specifically for any returned values but to demonstrate that the waveform processing does
+    # undeniably cause a consistent shift in the values returned. The "0" exception for processed is due to the nature of derivatives -- if two samples
+    # next to each other are identical, the derivative will be 0.
+    for processed, unprocessed in zip(processed_waveforms, unprocessed_waveforms):
+        assert abs(unprocessed) < 1
+        assert abs(processed) > 1 or processed == 0
+
+
+def test_measurement_stats_str(session):
+    test_voltage = 1.0
+    test_record_length = 1000
+    test_channels = 0
+    test_num_records = 1
+    session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
+    session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
+    with session.initiate():
+        measurement_stat = session.channels[test_channels].fetch_measurement_stats(niscope.enums.ScalarMeasurement.NO_MEASUREMENT, 5.0)
+
+    # As defined in the MeasurementStat class for string output:
+    row_format_g = '{:<20}: {:,.6g}\n'
+    row_format_d = '{:<20}: {:,}\n'
+    row_format_s = '{:<20}: {:}\n'
+    string_representation = ''
+    string_representation += row_format_s.format('channel', 0)
+    string_representation += row_format_d.format('record', 0)
+    string_representation += row_format_g.format('result', 0)
+    string_representation += row_format_g.format('mean', 0)
+    string_representation += row_format_g.format('standard deviation', 0)
+    string_representation += row_format_g.format('minimum value', 0)
+    string_representation += row_format_g.format('maximum value', 0)
+    string_representation += row_format_d.format('num in stats', 0)
```

Comment:
this is almost a re-implementation of  `__str__`, it would be clearest and simpler to simply hardcode and compare against the string.

Reply:
Hardcoding and comparing was my initial attempt at writing this test, but for whatever reason it simply didn't work, even when comparing to a string that was quite literally the output of the str function copy+pasted. It's possible I did something wrong, though, so I'll try it again and see what I can do.

Reply:
I believe all we did to test similar functionality for in other places was to call it to make sure it didn't error, but did not try to check the returned value, other than for maybe type. Not value though.